components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/SecurityQuestionApi.java

```diff
@@ -193,9 +194,12 @@ public InitiateAllQuestionResponse securityQuestionsGet(String username, String
             tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
         }
 
-        if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
-            basePath = IdentityManagementEndpointUtil.buildEndpointUrl("t/" + tenantDomain +
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            basePath = IdentityManagementEndpointUtil.buildTenantQualifiedUrl(
                     IdentityManagementEndpointConstants.UserInfoRecovery.RECOVERY_API_RELATIVE_PATH);
+        } else if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equalsIgnoreCase(tenantDomain)) {
+            basePath = IdentityManagementEndpointUtil.buildEndpointUrl("t/" + tenantDomain
```

Comment:
Can't we the buildTenantQualifiedUrl() method here too?

Reply:
It's already used here.

Reply:
What if we move this base path building logic to the util as a whole. Because even the tenant qualified url there is actually not building a tenant qualified url always.

The whole segment here is about building the base url of the service

Reply:
Updated the PR as discussed.